src/components/Pagination/Pagination.tsx

```diff
@@ -1 +1,84 @@
-export const Pagination = () => {};
+import { FC } from 'react';
+import classNames from 'classnames';
+import { getNumbers } from '../../utils/utils';
+
+interface PaginationProps {
+  total: number
+  perPage: number
+  currentPage: number
+  onPageChange: (currentPage: number) => void
+}
+
+export const Pagination: FC<PaginationProps> = ({
+  onPageChange,
+  currentPage,
+  perPage,
+  total,
+}) => {
+  const totalPageCount = Math.ceil(total / perPage);
+  const isFirstPageIndex = currentPage === 1;
+  const isLastPageIndex = currentPage === totalPageCount;
+  const paginationRange = getNumbers(1, totalPageCount);
+
+  const handlePreviousPageClick = () => {
+    return !isFirstPageIndex
+      && onPageChange(currentPage - 1);
```

Comment:
If you don't need to return anything - use simple `if` instead. It's easier to read and debug.